File: verify.js
const { syncPlaywright } = require('playwright');
const path = require('path');
const fs = require('fs');

async function verifyFeature() {
    // Create verification directories
    const verificationDir = path.join('/home/jules/verification');
    const videoDir = path.join(verificationDir, 'video');
    if (!fs.existsSync(verificationDir)) {
        fs.mkdirSync(verificationDir, { recursive: true });
    }
    if (!fs.existsSync(videoDir)) {
        fs.mkdirSync(videoDir, { recursive: true });
    }

    const browser = await require('playwright').chromium.launch({ headless: true });
    const context = await browser.newContext({
        recordVideo: {
            dir: videoDir,
            size: { width: 800, height: 600 }
        }
    });

    const page = await context.newPage();

    try {
        console.log("Navigating to game...");
        await page.goto("http://localhost:8080");
        await page.waitForTimeout(1000);

        console.log("Bypassing menu...");
        await page.evaluate(() => {
            window.game.scene.scenes[0].scene.start('GameScene', {levelConfig: GameConfig.LEVELS[0]});
        });

        console.log("In Game...");
        // Wait a bit to let the game start and arrow to show up
        await page.waitForTimeout(2000);

        // Take a screenshot of the game scene to verify player size
        const screenshotPath = path.join(verificationDir, 'verification.png');
        await page.screenshot({ path: screenshotPath });
        console.log("Screenshot saved to", screenshotPath);

        await page.waitForTimeout(1000);

    } catch (e) {
        console.error("Error during Playwright script:", e);
    } finally {
        await context.close();
        await browser.close();
        console.log("Verification script finished.");
    }
}

verifyFeature();

File: verify.py
from playwright.sync_api import Page, expect, sync_playwright

def verify_feature(page: Page):
    # Navigate to the game
    page.goto("http://localhost:8080")
    page.wait_for_timeout(1000)

    # Force the game to start into GameScene directly to skip the menus
    page.evaluate("window.game.scene.scenes[0].scene.start('GameScene', {level: 1, world: 1})")
    page.wait_for_timeout(2000) # Give the scene some time to load

    # Capture a screenshot showing the player sprite size
    page.screenshot(path="/home/jules/verification/verification.png")
    page.wait_for_timeout(1000)

if __name__ == "__main__":
    with sync_playwright() as p:
        browser = p.chromium.launch(headless=True)
        context = browser.new_context(record_video_dir="/home/jules/verification/video")
        page = context.new_page()
        try:
            verify_feature(page)
        finally:
            context.close()
            browser.close()